nomos-services/sdp/src/adapters/wallet/mod.rs

```diff
@@ -0,0 +1,35 @@
+pub mod mock;
+
+use nomos_core::{
+    mantle::{NoteId, SignedMantleTx, tx_builder::MantleTxBuilder},
+    sdp::{ActiveMessage, DeclarationMessage, WithdrawMessage, ZkPublicKey},
+};
+
+#[async_trait::async_trait]
+pub trait SdpWalletAdapter {
+    type Error;
+
+    // TODO: Pass relay when wallet service is defined.
+    fn new() -> Self;
```

Comment:
You can get the overwatch handle already here. Then the implementor decides what to relay to.

Reply:
I copied the same approach from dispersal, where we will provide a handle here when the actual service is ready. Does that makes sense?